discovery_engine_core/web-api2/src/personalization/routes.rs

```diff
@@ -40,33 +58,203 @@ pub(super) fn configure_service(config: &mut ServiceConfig) {
     config.service(scope);
 }
 
-//FIXME use actual body
-#[derive(Deserialize, Debug)]
-struct UpdateInteractions {}
+/// Represents user interaction request body.
+#[derive(Clone, Debug, Deserialize)]
+pub(crate) struct UpdateInteractions {
+    pub(crate) documents: Vec<UserInteractionData>,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+pub(crate) struct UserInteractionData {
+    #[allow(dead_code)]
+    #[serde(rename = "id")]
+    pub(crate) document_id: DocumentId,
+    #[allow(dead_code)]
+    #[serde(rename = "type")]
+    pub(crate) interaction_type: UserInteractionType,
+}
 
 async fn update_interactions(
-    config: Data<Config>,
+    state: Data<AppState>,
     user_id: Path<UserId>,
-    interactions: Json<UpdateInteractions>,
+    Json(interactions): Json<UpdateInteractions>,
 ) -> Result<impl Responder, Error> {
-    dbg!((config, user_id, interactions));
-    if true {
-        Err(Unimplemented {
-            functionality: "/users/{user_id}/interactions",
-        })?;
+    state.db.user_seen(&user_id).await?;
+
+    let ids = interactions
+        .documents
+        .iter()
+        .map(|document| &document.document_id)
+        .collect_vec();
+    let documents = state.elastic.get_documents_by_ids(&ids).await?;
+    let embeddings = documents
+        .into_iter()
+        .map(|document| (document.id, document.embedding))
+        .collect::<HashMap<_, _>>();
+
```

Comment:
:grey_exclamation:  No longer errors if some of the documents don't exist anymore.

There is always a race between us deleting documents and clients potentially still expecting them, and there is also nothing good the client can do based on that error.